ansible/roles/providers/templates/etc/udev/rules.d/66-azure-storage.rules

```diff
@@ -0,0 +1,34 @@
+# Azure specific rules.
+ACTION!="add|change", GOTO="walinuxagent_end"
+SUBSYSTEM!="block", GOTO="walinuxagent_end"
+ATTRS{ID_VENDOR}!="Msft", GOTO="walinuxagent_end"
+ATTRS{ID_MODEL}!="Virtual_Disk", GOTO="walinuxagent_end"
+
+# Match the known ID parts for root and resource disks.
+ATTRS{device_id}=="?00000000-0000-*", ENV{fabric_name}="root", GOTO="wa_azure_names"
+ATTRS{device_id}=="?00000000-0001-*", ENV{fabric_name}="resource", GOTO="wa_azure_names"
+
+# Gen2 disk.
+ATTRS{device_id}=="{f8b3781a-1e82-4818-a1c3-63d806ec15bb}", ENV{fabric_scsi_controller}="scsi0", GOTO="azure_datadisk"
+# Create symlinks for data disks attached.
+ATTRS{device_id}=="{f8b3781b-1e82-4818-a1c3-63d806ec15bb}", ENV{fabric_scsi_controller}="scsi1", GOTO="azure_datadisk"
+ATTRS{device_id}=="{f8b3781c-1e82-4818-a1c3-63d806ec15bb}", ENV{fabric_scsi_controller}="scsi2", GOTO="azure_datadisk"
+ATTRS{device_id}=="{f8b3781d-1e82-4818-a1c3-63d806ec15bb}", ENV{fabric_scsi_controller}="scsi3", GOTO="azure_datadisk"
+GOTO="walinuxagent_end"
+
+# Parse out the fabric n ame based off of scsi indicators.
+LABEL="azure_datadisk"
+ENV{DEVTYPE}=="partition", PROGRAM="/bin/sh -c 'readlink /sys/class/block/%k/../device|cut -d: -f4'", ENV{fabric_name}="$env{fabric_scsi_controller}/lun$result"
+ENV{DEVTYPE}=="disk", PROGRAM="/bin/sh -c 'readlink /sys/class/block/%k/device|cut -d: -f4'", ENV{fabric_name}="$env{fabric_scsi_controller}/lun$result"
+
+ENV{fabric_name}=="scsi0/lun0", ENV{fabric_name}="root"
+ENV{fabric_name}=="scsi0/lun1", ENV{fabric_name}="resource"
+# Don't create a symlink for the cd-rom.
+ENV{fabric_name}=="scsi0/lun2", GOTO="walinuxagent_end"
+
+# Create the symlinks.
+LABEL="wa_azure_names"
+ENV{DEVTYPE}=="disk", PROGRAM+="/bin/sh -c '[ -L /dev/disk/azure ] && unlink /dev/disk/azure'", SYMLINK+="disk/azure/$env{fabric_name}"
```

Comment:
After a VM is rebooted, the `/dev/disk/azure` is reseted back to `../sda`.
I am unable to figure out which process creates this link. I have ruled out `systemd-udevd`, `cloud-init` and `waagent` services with some testing. 

Following is the hack for working around unlinking the file.
created https://d2iq.atlassian.net/browse/D2IQ-96374 to remove the hack in future.

moving logic to remove symlinks in udev rules `PROGRAM+="/bin/sh -c '[ -L /dev/disk/azure ] && unlink /dev/disk/azure'"` to ensure that rules run upon reboots.